source/common/upstream/health_checker_base_impl.cc

```diff
@@ -59,12 +60,18 @@ std::chrono::milliseconds HealthCheckerImplBase::interval(HealthState state,
   if (cluster_.info()->stats().upstream_cx_total_.used()) {
     switch (state) {
     case HealthState::Unhealthy:
-      base_time_ms = changed_state == HealthTransition::Changed ? unhealthy_edge_interval_.count()
-                                                                : unhealthy_interval_.count();
+      base_time_ms = (changed_state == HealthTransition::Changed ||
+                      changed_state == HealthTransition::ChangePending) &&
+                             state_run == 1
```

Comment:
Looking at the documentation of HAProxy, I don't think this is quite right. The edge-interval should be used anytime the last health check result doesn't agree with the current state. Eg if the thresholds are 5, and the state is unhealthy, and I get a success from the last health check, the next 4 health checks should use the edge interval (assuming they all pass). Not just 1 time.

Reply:
Hmm, interesting. Can you point to the documentation that mentions this behavior? I don't recall seeing this.

Reply:
https://www.haproxy.com/documentation/aloha/7-0/traffic-management/lb-layer7/health-checks/

The original PR mentioned trying to replicate HAProxy's behavior, so I'm assuming that's what we should be doing here.

Reply:
You're correct. I requested the link because I'm just looking for the reference documentation to ensure the behavior I follow is correct. Thanks for providing that.